code/player/Player.UI.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Sandbox;
+using Cinema.UI;
+
+namespace Cinema;
+
+public partial class Player
+{
+    /// <summary>
+    /// On the server and client, this will be true if the client has a menu open. 
+    /// </summary>
+    [ClientInput]
+    public bool IsMenuOpen { get; set; }
```

Comment:
The setters for these two properties should be `protected` or `private` since `OpenMenu`/`CloseMenu` are the mutation functions for these